syntax = "proto3";

package postservice;

message Post {
  int64 id = 1;
  int64 user_id = 2;
  string title = 3;
  string content = 4;
}

message CreatePostRequest {
  int64 user_id = 1;
  string title = 2;
  string content = 3;
}

message UpdatePostRequest {
  int64 id = 1;
  int64 user_id = 2;
  string title = 3;
  string content = 4;
}

message DeletePostRequest {
  int64 id = 1;
  int64 user_id = 2;
}

message GetPostRequest {
  int64 id = 1;
}

message ListPostsRequest {
  int64 page_number = 1;
  int64 posts_per_page = 2;
}

message ListPostsResponse {
  repeated Post posts = 1;
  int64 posts_count = 2;
}

service PostService {
  rpc CreatePost (CreatePostRequest) returns (Post);
  rpc GetPost (GetPostRequest) returns (Post);
  rpc UpdatePost (UpdatePostRequest) returns (Post);
  rpc DeletePost (DeletePostRequest) returns (Post);
  rpc ListPosts (ListPostsRequest) returns (ListPostsResponse);
}
